api/v1/nonclusterhost_types.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2024 Tigera, Inc. All rights reserved.
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NonClusterHostSpec enables non-cluster hosts to connect to a cluster.
+type NonClusterHostSpec struct {
+	// Location of the log ingestion point for non-cluster hosts. example: https://1.2.3.4:443
+	// +kubebuilder:validation:Pattern=`^https://.+$`
+	Endpoint string `json:"endpoint"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster
+
+// NonClusterHost installs the components required for non-cluster host log collection.
+// At most one instance of this resource is supported. It must be named "tigera-secure".
+type NonClusterHost struct {
```

Comment:
As far as naming goes, I think we should consider something else.  Since k8s APIs are typically representative of an object, this naming implies that this object represents a single non-cluster host (i.e., there would be one per host) whereas I think this is actually just general configuration. 

What other types of configuration do you foresee us putting in this resource? If it's just a log storage endpoint, then perhaps it belongs in the LogStorage resource? But I suspect there is more planned for this.

Reply:
We spoke offline and I think we will proceed with the current name.